### ISSUE TITLE
Feature Suggestion: 'split'

### ISSUE THREAD

**Issue by DarrenTsung** [rcoh/angle-grinder#83]
Hi, thanks for the great tool!

One common case that I run into quite often is logs with the format like:
```
Oct 09 20:22:21 web-001 influxd[188053]: [httpd] 127.0.0.1 - - [09/Oct/2019:20:22:21 +0000] "POST /write HTTP/1.0" 204 0 "-" "reqwest/0.9.13" 7g5f9126-ead2-11e
```

And wanting to analyze the log based on one of the columns like: `"reqwest/0.9.13"`. To do so I usually do something like:
```
agrind '* | parse "* * * * *: * * - - [*] \"*\" * * \"*\" \"*\" * *" as a, b, c, d, e, f, g, h, i, j, k, l, request_header, n | count by request_header'
```

This is kinda painful, so I was going to suggest a feature like:
```
// -2 being the second last field
agrind '* | split " " | count by -2'
```

Now that I'm writing this out, I realize that a `split` operator would have to dynamically change the parsing / column mappings based on the lines coming in.. which I don't think is supported with the current functionality.

Actually, the `json` operator dynamically parses the line, so maybe this can be supported in a similar way.

It might be difficult to support -index column naming, my thoughts were that the same data could map to both `column1` and `column-8` or something. But not sure how realistic that is.

Additional thoughts: it would be nice if the `split` operator worked with quoting so that `split ","` grouped `abc,def,"g,h"` as `abc`, `def`, `g,h`.

---

Thoughts on:
1. Is this feature a good idea to add?
2. How feasible is each part of the feature with the current architecture?
3. Would you be open to a PR with this functionality? (no promises)

**Comment by rcoh** [comment]
generally splitting on spaces or commans etc. is pretty reasonable. I would support this by:
1. `split` actually just parses into an `Data::Array`. I can't remember if I landed support for that yet, but I can do it this week.
2. You can then just access it with array access (which I know I didn't land yet) so that instead of having fake columns you can just do `row[-1]`

The thing that doesn't quite fit into the current machinery would be having an array as a top level record. The quickest way to handle that would be to do something like:

`split as key_name | sum(key_name[1])` (`key_name`) becomes the implicit top-level key holding the result of the split operator. Maybe by default it goes to `_split`? 

All of this said, the proposal to add a way to have predefined operators is possibly the most generally useful thing for your use case), and probably a little easier to tackle since it has less interaction with the data pipeline internals (#81 )

**Comment by rcoh** [comment]
I've added support for `data::Value::Array` -- this should make the split operator very straightforward to add.

**Comment by DarrenTsung** [comment]
Nice! I will try to open a PR to add the split operator.

**Comment by rcoh** [comment]
Fixed in #86
